features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
I will update the PR to take advantages of the changes from @bridadan.

As a side note: 

> Is this guaranteed by the C++ standard to be initialized on first use or could it also be initialized at boot? 

Yes it is guaranteed to be initialized on first use.

> lso, I doubt this is thread safe for GCC, although if utest calls this in its initialization sequence then it will be initialized in a single threaded manner. 

No its not thread safe in C++03 (it is in C++11) but it can be thread safe for abi conforming to the itanium ABI: https://mentorembedded.github.io/cxx-abi/abi.html#once-ctor

It is possible to overload these definitions: 
* `extern "C" int __cxa_guard_acquire ( __int64_t *guard_object );`
* `extern "C" void __cxa_guard_release ( __int64_t *guard_object );`
* `extern "C" void __cxa_guard_abort ( __int64_t *guard_object );`

because the ARMEABI is based on itanium ABI and therefore it is possible to have thread safe static objects constructions on mbed without any workaround.